fhir-smart/src/main/java/com/ibm/fhir/smart/AuthzPolicyEnforcementPersistenceInterceptor.java

```diff
@@ -114,26 +122,52 @@ public void beforeSearch(FHIRPersistenceEvent event) throws FHIRPersistenceInter
 
             if (compartment != null) {
                 // Compartment search - validate patient access
-                if (!"Patient".equals(compartment)) {
-                    String msg = "Compartment search with compartment '" + compartment + "' is not permitted.";
-                    throw new FHIRPersistenceInterceptorException(msg).withIssue(FHIRUtil.buildOperationOutcomeIssue(msg, IssueType.FORBIDDEN));
-                }
-                if (!patientIdFromToken.contains(compartmentId)) {
-                    String msg = "Interaction with 'Patient/" + compartmentId + "' is not permitted under patient context '" + patientIdFromToken + "'.";
+                switch(compartment) {
+                case "Patient":
+                    if (!patientIdFromToken.contains(compartmentId)) {
+                        String msg = "Interaction with 'Patient/" + compartmentId + "' is not permitted under patient context '" + patientIdFromToken + "'.";
+                        throw new FHIRPersistenceInterceptorException(msg).withIssue(FHIRUtil.buildOperationOutcomeIssue(msg, IssueType.FORBIDDEN));
+                    }
+                    try {
+                        // TODO: confirm this entire try-catch block is safe to remove
```

Comment:
Has this TODO been confirmed? Can this block be removed?

Reply:
I've now confirmed this and will remove this block in my next commit.
The error I get back on a search like `GET [base]/Patient/123/MedicationKnowledge` (with or without this interceptor) is
```
Invalid resource type: MedicationKnowledge for compartment: Patient
```

Reply:
done